### ISSUE TITLE
Enhance Accessibility of Avatar Components with role="img" Attribute

### ISSUE THREAD

**Issue by georgewrmarshall** [MetaMask/metamask-extension#23251]
### What is this about?

To improve the accessibility of our Avatar components, especially when a screen reader is used, we propose adding the role="img" attribute to these components. This enhancement aims to prevent screen readers from reading out the fallback letter, providing a more meaningful and contextually appropriate description. For instance, when tabbing through networks in the extension, the screen reader should announce "S, Sepolia" instead of just "S". This change will ensure our UI is more accessible and user-friendly for individuals relying on screen readers.

https://github.com/MetaMask/metamask-extension/assets/8112138/ec6ab28f-8fbf-4cd2-b0ea-787f82d7bcae

See role on Chakra avatar

<img width="1135" alt="Screenshot 2024-02-29 at 12 17 43 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/7df763bf-a525-4679-9d55-e0b3551433e8">

### Technical Details

Scope: Update all Avatar components across the application.
Implementation:
- Add the role="img" attribute to the Avatar component's root element.
- Ensure that the aria-label attribute is dynamically set to convey the correct context, such as the network name in the case of network avatars.

### Acceptance Criteria

- The role="img" attribute is present on all Avatar components.
- Screen readers correctly interpret the Avatar component, reading out the correct information (e.g., "Sepolia") instead of including the the fallback letter (e.g., "S Sepolia")".
- There is no regression in the functionality or appearance of the Avatar components.
- Accessibility testing confirms that the update improves the screen reader experience.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)
